freva-rest/entrypoint.sh

```diff
@@ -0,0 +1,143 @@
+        if mongosh --quiet --eval "db.adminCommand('ping')" &>/dev/null; then
+            return 0
+        fi
+        sleep 1
+        log_debug "Waiting for MongoDB to start... (attempt $i/30)"
+    done
+    return 1
+}
+
+reset_mongo_user() {
+    log_info "Resetting MongoDB user..."
+    mongosh admin --quiet --eval "
+        if (db.getUser('${MONGO_USER}')) {
+            db.dropUser('${MONGO_USER}');
+            print('Existing user dropped');
+        } else {
+            print('User does not exist, proceeding with creation new user');
+        }
+        db.createUser({
+            user: '${MONGO_USER}',
+            pwd: '${MONGO_PASSWORD}',
+            roles: [
+                { role: 'root', db: 'admin' },
+                { role: 'userAdminAnyDatabase', db: 'admin' },
+                { role: 'dbAdminAnyDatabase', db: 'admin' },
+                { role: 'readWriteAnyDatabase', db: 'admin' }
+            ]
+        });"
+    if [ $? -eq 0 ]; then
+        log_info "MongoDB user reset successfully"
+    else
+        log_error "Failed to reset MongoDB user"
+    fi
+}
+
+verify_auth() {
+    log_debug "Verifying authentication..."
+    mongosh admin --quiet --eval "
+        try {
+            db.auth('${MONGO_USER}', '${MONGO_PASSWORD}');
+            db.adminCommand('listDatabases');
+            quit(0);
+        } catch(err) {
+            quit(1);
+        }"
+}
+
+init_mongodb() {
+    log_service "=== Initializing MongoDB ==="
+    log_info "Starting MongoDB without authentication..."
+    mongod --dbpath ${MONGO_HOME}/data --port ${MONGO_PORT} --bind_ip_all --fork --logpath "$LOG_FILE" --noauth &
+    wait_for_mongo
+    if ! verify_auth; then
+        log_warn "Authentication failed with existing credentials - resetting user..."
+        reset_mongo_user
+    else
+        log_info "Existing credentials are valid"
+    fi
+    log_info "Shutting down MongoDB for restart..."
+    mongod --shutdown --dbpath ${MONGO_HOME}/data
+    sleep 5
+    log_info "Starting MongoDB with authentication..."
+    mongod --dbpath ${MONGO_HOME}/data --port ${MONGO_PORT} --bind_ip 0.0.0.0 --auth &
+    wait_for_mongo
+    log_info "Initializing MongoDB collections..."
+    mongosh --authenticationDatabase admin -u "${MONGO_USER}" -p "${MONGO_PASSWORD}" --eval "
+        if (db.getSiblingDB('${MONGO_DB}').getCollection('searches').countDocuments() == 0) {
+            db.getSiblingDB('${MONGO_DB}').createCollection('searches');
+        }
+    "
+}
+
+init_solr() {
+    log_service "=== Initializing Solr ==="
+    solr start -force
+    until curl -s "http://localhost:${SOLR_PORT}/solr/admin/ping" >/dev/null 2>&1; do
+        log_debug "Waiting for Solr to start..."
+        sleep 1
+    done
+    log_info "Solr started successfully"
+}
+
+main() {
+    display_logo
+    log_service "Initializing services..."
+    init_mongodb
+    init_solr
+    log_service "Starting freva-rest..."
+    exec python3 -m freva_rest.cli
```

Comment:
Two things to consider:

- I think you want to allow some flexibility. for example not starting apache solr, mongo db etc. For this you could use env variables like `USE_SOLR=1` or similar.
- I believe `python3 -m freva_rest.cli` should be the cmd? Then also it would be nice to add some checks like tin the mariadb entrypoint whether the cmd containes only flags like `--port 8080`. In this case the cmd is not `--port 8080` but add to `python3 -m freva_rest.cli`